pandas/tests/indexes/base_class/test_constructors.py

```diff
@@ -29,7 +30,8 @@ def test_construction_list_mixed_tuples(self, index_vals):
     def test_constructor_wrong_kwargs(self):
         # GH #19348
         with pytest.raises(TypeError, match="Unexpected keyword arguments {'foo'}"):
-            Index([], foo="bar")
+            with tm.assert_produces_warning(FutureWarning):
+                Index([], foo="bar")
```

Comment:
might be slightly tricky but can you check based on the index construction fixture so *all* klasses are convered for checking an in valid keyword shows the deprecation?

Reply:
None of the others accept kwargs, except for PeriodIndex which checks them (though #38615)